C-Sharp.md

```diff
@@ -1,6 +1,232 @@
+
+For complete list of all built-in data types see [here](https://msdn.microsoft.com/en-us/library/ms228360)
+
+
+ [*_Reference types_*](https://msdn.microsoft.com/en-us/library/490f96s2.aspx) : Variables of reference types store references to their objects, which means they store the address to the location of data on the [stack](http://gribblelab.org/CBootcamp/7_Memory_Stack_vs_Heap.html#orgheadline2), also known as pointers. Actual data is stored on the [heap](http://gribblelab.org/CBootcamp/7_Memory_Stack_vs_Heap.html#orgheadline3) memory.
+ Assigning reference type to another doesn't copy the data, instead it creates the second copy of reference which points to the same location on the heap.
+
+###Flow control statements
+
+* [If else][ifelse] statement
+```csharp
+            int myScore = 700;
+            if (myScore == 700)
+            {
+                Console.WriteLine("I get printed on the console");
+            }
+            else if (myScore > 10)
+            {
+                Console.WriteLine("I don't");
+            }
+            else
+            {
+                Console.WriteLine("I also don't");
+            }
+```
+
+```csharp
+            /** Ternary operators
+             A simple if/else can also be written as follows
+             <condition> ? <true> : <false> **/
+            int myNumber = 10;
+            string isTrue = myNumber == 10 ? "Yes" : "No";
+
+```
+* [Switch][switch] statement
+
+```csharp
+int myNumber = 3;
+        switch (myNumber)
+        {
+            // A switch section can have more than one case label.
+            case 0:
+            case 1:
+                Console.WriteLine("Case 0 or 1");
+              /** Most switch sections contain a jump statement, such as a break, goto, or return. **/
+                break;
+            case 2:
+                Console.WriteLine("Case 2");
+                break;
+            // 7 - 4 in the following line evaluates to 3.
+            case 7 - 4:
+                Console.WriteLine("Case 3");
+                break;
+            /** If the value of myNumber is not 0, 1, 2, or 3 the
+            default case is executed.**/
+            default:
+                Console.WriteLine("Default case. This is also optional");
+              break;// could also throw new Exception() instead
+        }
+```
+* [For][for] & [Foreach][foreach]
+
+```csharp
+
+    for (int i = 0; i < 10; i++)
+  	{
+  	    Console.WriteLine(i); //prints  0-9
+  	}
+    for (int i = 0; i <= 10; i++)
+  	{
+  	    Console.WriteLine(i); //prints  0-10
+  	}
+    for (int i = 10 - 1; i >= 0; i--) //decrement loop
+  	{
+  	    Console.WriteLine(i);//prints 9-0
+  	}
+    for (; ; )
+      {
+           /** All of the expressions are optional. This statement
+           creates an infinite loop.**/
+      }
+
+    int[] array = new int[] { 0, 1, 1, 2, 3, 5, 8, 13 }; //declares array of integer values
+      foreach (int item in array)
```

Comment:
It is probably worth talking about enumerable collections at this point - that's the real value of the `foreach`